c2corg_api/views/user.py

```diff
@@ -288,6 +296,146 @@ def post(self):
         return {}
 
 
+class UpdateAccountSchema(colander.MappingSchema):
+    email = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.All(
+                colander.Email(),
+                colander.Function(
+                    partial(available_user_attribute, 'email'),
+                    'Already used email'
+                )
+            ))
+    toponame = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.Length(min=3))
+    forumname = colander.SchemaNode(
+            colander.String(),
+            missing=colander.drop,
+            validator=colander.All(
+                colander.Length(min=3),
+                colander.Function(
+                    partial(available_user_attribute, 'forum_username'),
+                    'Already used forum name'
+                )
+            ))
+    currentpassword = colander.SchemaNode(
+            colander.String(encoding=ENCODING),
+            validator=colander.Length(min=3))
+    newpassword = colander.SchemaNode(
+            colander.String(encoding=ENCODING),
+            missing=colander.drop,
+            validator=colander.Length(min=3))
+
+
+@resource(path='/users/account', cors_policy=cors_policy)
+class UserAccountRest(object):
+    updateschema = UpdateAccountSchema()
+
+    def __init__(self, request):
+        self.request = request
+
+    def get_user(self):
+        userid = self.request.authenticated_userid
+        return DBSession.query(User).get(userid)
+
+    @restricted_view(renderer='json', http_cache=0)
+    def get(self):
+        user = self.get_user()
+        return {
```

Comment:
Ok, so it's an error that `name` is not required. How does it work during the registration? For example do we expect that the user enters three different user names? Or only require `user_name` and use this name for the other two if no value was given?